doc/modules/preprocessing.rst

```diff
@@ -593,6 +593,64 @@ the 2 features with respectively 3 categories each.
 See :ref:`dict_feature_extraction` for categorical features that are
 represented as a dict, not as scalars.
 
+.. _one_hot_encoder_infrequent_categories:
+
+Infrequent categories
+---------------------
+
+:class:`OneHotEncoder` supports outputing a feature that combines infrequent
+categories in the training data. For each input feature that has an infrequent
+category a new column is formed to represent it. The parameters to enable the
+gathering of infrequent categories are `min_frequency` and `max_categories`.
+
+1. `min_frequency` is either an  integer greater or equal to 1, or a float in
+the interval `(0.0, 1.0)`. If `min_frequency` is an integer, categories with a
+cardinality smaller than `min_frequency * n_samples`  will be considered
+infrequent. If `min_frequency` is a float, categories with a cardinality smaller
+than this fraction of the total number of samples will be considered infrequent.
+
+2. `max_categories` is either `None` or any integer greater than 1. This
+parameter sets an upper limit to the number of output features for each input
+feature. `max_categories` includes the feature that combines infrequent
+categories.
+
+In the following example, the categories, `'dog', 'snake'` are considered
+infrequent::
+
+   >>> X = np.array([['dog'] * 5 + ['cat'] * 20 + ['rabbit'] * 10 + 
+   ...               ['snake'] * 3]).T
+   >>> enc = preprocessing.OneHotEncoder(min_frequency=6,
+   ...                                   handle_unknown='auto').fit(X)
+   >>> enc.transform([['dog'], ['cat'], ['rabbit'], ['snake']]).toarray()
+   array([[0., 0., 1.],
+          [1., 0., 0.],
+          [0., 1., 0.],
+          [0., 0., 1.]])
+
+By setting handle_unknown to `'auto'`, unknown categories will be considered
+infrequent::
+
+   >>> enc.transform([['dragon']]).toarray()
+   array([[0., 0., 1.]])
+
+:meth:`OneHotEncoder.get_feature_names` uses 'infrequent' as the infrequent
+feature name::
+
+   >>> enc.get_feature_names()
+   array(['x0_cat', 'x0_rabbit', 'x0_infrequent'], dtype=object)
+
+Infrequent categories can be filtered out using `min_frequency` and
+`max_categories`. In the following example, we set `max_categories=2` to
+limit the number of features in the output::
```

Comment:
This will result in all but the 'cat' category to be considered infrequent, leading to two features, one for ``'cat'`` and one for infrequent categories - which are all the others.